improver/standardise.py

```diff
@@ -105,9 +105,126 @@ def grid_contains_cutout(grid, cutout):
+        """
+        Modify input cube in place to conform to mandatory dtype and unit
+        standards.
+
+        Args:
+            cube (iris.cube.Cube:
+                Cube to be updated in place
+
+        """
+
+        def as_correct_dtype(obj, required_dtype):
+            """
+            Returns an object updated if necessary to the required dtype
+
+            Args:
+                obj (np.ndarray):
+                    The object to be updated
+                required_dtype (np.dtype):
+                    The dtype required
+
+            Returns:
+                np.ndarray
+            """
+            if obj.dtype != required_dtype:
+                return obj.astype(required_dtype)
+            return obj
+
+        cube.data = as_correct_dtype(cube.data, get_required_dtype(cube))
+        for coord in cube.coords():
+            if coord.name() in TIME_COORDS and not check_units(coord):
+                coord.convert_units(get_required_units(coord))
+            req_dtype = get_required_dtype(coord)
+            # ensure points and bounds have the same dtype
+            if np.issubdtype(req_dtype, np.integer):
+                coord.points = np.around(coord.points)
+            coord.points = as_correct_dtype(coord.points, req_dtype)
+            if coord.has_bounds():
+                if np.issubdtype(req_dtype, np.integer):
+                    coord.bounds = np.around(coord.bounds)
+                coord.bounds = as_correct_dtype(coord.bounds, req_dtype)
+
+    def process(
+        self,
+        cube,
+        new_name=None,
+        new_units=None,
+        coords_to_remove=None,
+        attributes_dict=None,
+    ):
+        """
+        Perform compulsory and user-configurable metadata adjustments
+
+        Args:
+            cube (iris.cube.Cube):
+                Input cube to be standardised
+            new_name (str or None):
+                Optional rename for output cube
+            new_units (str or None):
+                Optional unit conversion for output cube
+            coords_to_remove (list of str or None):
+                Optional list of scalar coordinates to remove from output cube
+            attributes_dict (dict or None):
+                Optional dictionary of required attribute updates. Keys are
+                attribute names, and values are the required value or "remove".
+
+        Returns:
+            iris.cube.Cube
+        """
+        # standard metadata updates
+        cube = self._collapse_scalar_dimensions(cube)
+        self._standardise_dtypes_and_units(cube)
+
+        # optional metadata updates
+        if new_name:
+            cube.rename(new_name)
+        if new_units:
+            cube.convert_units(new_units)
+        if coords_to_remove:
+            self._remove_scalar_coords(cube, coords_to_remove)
+        if attributes_dict:
+            amend_attributes(cube, attributes_dict)
+
+        return cube
+
+
+class RegridLandSea(BasePlugin):
```

Comment:
Are you planning to keep this plugin in this file or separate out the standardise metadata and regridding plugins into separate files?

Reply:
Planning to keep this in the same file, unless there's a strong preference otherwise.  It's all standardisation related, since we're normally using these plugins to put everything onto a "standard grid" - which is an underpinning concept for the system so I wouldn't say it's too Met Office specific to keep that way,

Reply:
I don't have a strong preference. I'm to leave them together. The only other way to do it is to move one into metadata and the other one somewhere else.

Reply:
Thank you